cronet/src/main/java/io/grpc/cronet/CronetClientTransport.java

```diff
@@ -151,13 +151,14 @@ public void run() {
     return new StartCallback().clientStream;
   }
 
-  @SuppressWarnings("GuardedBy")
   @GuardedBy("lock")
   private void startStream(CronetClientStream stream) {
-    streams.add(stream);
-    // TODO(b/145386688): This access should be guarded by 'stream.transportState().lock'; instead
-    // found: 'this.lock'
-    stream.transportState().start(streamFactory);
+    synchronized (lock) {
+      streams.add(stream);
+      synchronized (stream.transportState().lock) {
```

Comment:
stream.transportState().lock is never accessed outside of synchronized(lock) so this change should also not be needed.